### PR TITLE
Adds QDEL_NULL to qdel and null a variable

### PR DESCRIPTION
**Pull request by Cyberboss**
This is necessary

### CODE REVIEW

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -1286,6 +1286,7 @@ proc/pick_closest_path(value, list/matches = get_fancy_list_of_atom_types())
 #define RANDOM_COLOUR (rgb(rand(0,255),rand(0,255),rand(0,255)))
 
 #define QDEL_IN(item, time) addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, item), time, TIMER_STOPPABLE)
+#define QDEL_N(item) qdel(item); item = null
```
Comment (ChangelingRain) [inline code/__HELPERS/unsorted.dm]: Name this something more obvious instead of it being the above minus a letter, please.

**Comment by MrPerson** [comment]
Wouldn't it make more sense to have this be the default behavior? When would you ever want to call qdel() and not null the variable?

**Comment by Cyberboss** [comment]
That would require a refactor going through every invocation of qdel. I don't think ANYONE wants to handle that.

**Comment by MrPerson** [comment]
Why would it? Any instance where someone is deleting something and then continuing to use the reference is an error and should be treated as such

**Comment by MrPerson** [comment]
Although maybe list references would break if fed into the macro wrong

**Comment by Cyberboss** [comment]
The problem is a lot of invocations of `qdel(src)` continue to do things with src vars they know won't be touched in Destroy. Besides, adding an extra null out to every single qdel is not really necessary. This is mainly for qdel's in Destroy, that NEED to null stuff out

**Comment by MrPerson** [comment]
If you have a reference to an object, it fits one of five criteria:

1. The reference is local and based on a locate(), for(), or something along those lines. Nulling doesn't matter since the reference is temporary.
2. The reference is an argument passed through to the function. Nulling doesn't matter since the reference is temporary.
3. The reference is an entry in a list. The reference needs to be nulled.
4. The reference is a var local to the object itself or a global reference. The reference needs to be nulled.
5. The reference is src. Nulling doesn't matter since the reference is weak.

3-4 are basicallyt he same, 2 and 5 are just good behavior to get into where you do a qdel() last, and 1 is generally irrelevant.

**Pull request merged**